core/commands/files/files.go

```diff
@@ -471,7 +473,12 @@ Example:
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
-		defer fi.Close()
+
+		if noflush {
+			defer fi.Sync()
+		} else {
+			defer fi.Close()
+		}
```

Comment:
> MUST be safe and not leave the dag in an unrecoverable state.

It does, there wont be any corruption in that case as updates are performed only when they are complete.

> SHOULD be consistent and keep the result of the first write (journaled)

When you start throwing power outages and kill -9's around, you immediately lose the guarantee that you can keep *all* of your data. You can likely retain any data thats made it to disk, but programs arent written (even in the best of enterprise software stacks) to perfectly protect data no matter what. They get that guarantee by saying 'Dont run kill -9' and 'put your computer on battery backup'.

[If you start hard crashing programs...](https://i.imgflip.com/v2s24.jpg)

I do however agree that we should start a journal